src/pages/en/core-concepts/framework-components.md

```diff
@@ -59,33 +59,33 @@ import MyReactComponent from '../components/MyReactComponent.jsx';
 </html>
 ```
 
-By default, your framework components will render as static HTML. This is useful for templating components that are not interactive and avoids sending any unnecessary JavaScript to the client.
+By default, your framework components will only render on the server, as static HTML. This is useful for templating components that are not interactive and avoids sending any unnecessary JavaScript to the client.
 
 ## Hydrating Interactive Components
 
-A framework component can be made interactive (hydrated) using one of the `client:*` directives. This is a component attribute to define how your component should be **rendered** and **hydrated**.
+A framework component can be made interactive (hydrated) using a [`client:*` directive](/en/reference/directives-reference/#client-directives). These are component attributes that determine when your component's JavaScript should be sent to the browser.
 
-This [client directive](/en/reference/directives-reference/#client-directives) describes whether or not your component should be rendered at build-time, and when your component's JavaScript should be loaded by the browser, client-side.
-
-Most directives will render the component on the server at build time. Component JS will be sent to the client according to the specific directive. The component will hydrate when its JS has finished importing.
+With most directives, your component will first render on the server to generate static HTML. Component JavaScript will be sent to the browser according to the directive you chose. The component will then hydrate and become interactive.
 
 ```astro title="src/pages/interactive-components.astro" /client:\S+/
 ---
 // Example: hydrating framework components in the browser.
 import InteractiveButton from '../components/InteractiveButton.jsx';
 import InteractiveCounter from '../components/InteractiveCounter.jsx';
+import InteractiveModal from "../components/InteractiveModal.svelte"
 ---
 <!-- This component's JS will begin importing when the page loads -->
 <InteractiveButton client:load />
 
 <!-- This component's JS will not be sent to the client until
 the user scrolls down and the component is visible on the page -->
 <InteractiveCounter client:visible />
+
+<!-- This component won't render on the server, but will render on the client when the page loads -->
```

Comment:
I think this information is too important to be left in a code-block comment. Maybe we can separate them into client and server-rendered components?

Reply:
Yeah, as I was working on this PR, I felt the awkwardness of explaining `client:only` along with the other client directives. I agree that ideally we should split this up, but I'm not sure if it should happen in this PR.